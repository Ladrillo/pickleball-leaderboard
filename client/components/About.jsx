import React from 'react';
import Router from 'react-router/BrowserRouter';
import Match from 'react-router/Match';
import Link from 'react-router/Link';


const About = ({ pathname }) => {
    return (
        <div>
            <h1>ABOUT PAGE</h1>
            <br/>

            <a href = '/'>back home</a><br/>
            <a href = '/about'>About</a><br/>

            <Link to = { `${pathname}/consensus` }>Consensus</Link><br/>
            <Link to = { `${pathname}/dev_team` }>Dev Team</Link><br/>
            <Link to = { `${pathname}/fish_tank` }>Fish Tank</Link>

            <hr/>

            <Match pattern = { `${pathname}/:topicId` } component = { Topic } />
            <Match exactly pattern = { `${pathname}` } render = { () => <h2>Select Topic</h2> } />
        </div>
    );
};


const Topic = ({ params }) => {

    return (
        <div>
            <h2>{ params.topicId }</h2>
        </div>
    );
};

export default About;