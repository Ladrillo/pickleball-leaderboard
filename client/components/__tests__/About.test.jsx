import React from 'react';
import renderer from 'react-test-renderer';
import About from '../About';


describe(`About component`, () => {

	it(`Renders correctly...`, () => {
		const component = renderer.create(
			<About />
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});