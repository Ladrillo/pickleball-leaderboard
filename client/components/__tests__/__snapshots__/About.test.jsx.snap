exports[`About component Renders correctly... 1`] = `
<div>
  This is the about page
</div>
`;
